diff --git a/src/carburetor/ui/main.ui b/src/carburetor/ui/main.ui
index 7c096b5..e2803be 100644
--- a/src/carburetor/ui/main.ui
+++ b/src/carburetor/ui/main.ui
@@ -21,6 +21,7 @@ along with Carburetor.  If not, see <http://www.gnu.org/licenses/>. -->
   <requires lib="gtk" version="4.12"/>
   <requires lib="libadwaita" version="1.4"/>
   <object class="AdwApplicationWindow" id="MainWindow">
+    <property name="hide-on-close">True</property>
     <property name="icon-name">io.frama.tractor.carburetor</property>
     <signal name="realize" handler="on_mainpage_realize"/>
     <child>
